rpc/server_options.go

```diff
@@ -143,3 +157,25 @@ func WithAuthHandler(forType CredentialsType, handler AuthHandler) ServerOption
 		return nil
 	})
 }
+
+// WithAuthenticateToHandler returns a ServerOption which adds an authentication
+// handler designed to allow the caller to authenticate itself to some other entity.
+// This is useful when externally authenticating as one entity for the purpose of
+// getting access to another entity. Only one handler can exist and the forType
+// parameter will be the type associated with the JWT made for the authenticated to entity.
+// This can technically be used internal to the same server to "assume" the identity of
+// another entity but is not intended for such usage.
+func WithAuthenticateToHandler(forType CredentialsType, handler AuthenticateToHandler) ServerOption {
```

Comment:
made it simpler for users to get AuthenticateTo behavior and not mess it up as easily :). Note this is almost like an AWS assume role now but not designed for it (yet?)